CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+    commit messages and adhere to them. It makes the lives of those who come
+    after you a lot easier.
+
+
+
+Development Environments
+========================
+
+There are two environments, available on Linux and macOS, that you can use to
+develop Apache Airflow:
+
+-   `Local virtualenv development environment <#local-virtualenv-development-environment>`_
+    that supports running unit tests and can be used in your IDE.
+
+-   `Breeze Docker-based development environment <#breeze-test-development-environment>`_
+
+The table below summarizes differences between the two environments:
+
+
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+| **Development Environment** | **Test Coverage**                                                  | **Setup**                      |
++=============================+====================================================================+================================+
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+| Local virtualenv            | - (-) limited to unit tests with no dependencies                   | - (-) manual setup             |
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+| Breeze environment          | - (+) supports integration and unit tests                          | - (+) automated setup          |
+|                             | - (+) reproducible in other dev environments and Travis CI builds  | - (-) consumes docker space    |
+|                             |                                                                    | - (-) high CPU utilization     |
++-----------------------------+--------------------------------------------------------------------+--------------------------------+
+
+Typically, you are recommended to use both of these environments depending on
+your needs.
+
+Local virtualenv Development Environment
+----------------------------------------
+
+All details about using and running local virtualenv enviroment for Airflow can
+be found
+
+in `LOCAL_VIRTUALENV.rst <LOCAL_VIRTUALENV.rst>`__.
+
+Benefits:
+
+-   Packages are installed locally. No container environment is required.
+
+-   You can benefit from local debugging within your IDE.
+
+-   You can retain access to a Python virtualenv in your local IDE that meets
+    necessary requirements and supports autocompletion and running tests
+    directly from the IDE.
+
+Limitations:
+
+-   You have to maintain your dependencies and local environment consistent with
+    other development environments that you have on your local machine.
```

Comment:
Elena: //didn't  get this. my local devenv and other environments? such as?
Jarek: When you install dependencies in main python library without virtualenv, you can install dependencies in main "python" and then it might interfere with your all local virtualenvs. You can have several of those environments. You can easily skip this here. This is well-known fact among python developers. We can easily remove this line I think.